optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/Joiners.java

```diff
@@ -89,6 +109,18 @@
         return new SingleBiJoiner<>(leftMapping, JoinerType.GREATER_THAN_OR_EQUAL, rightMapping);
     }
 
+    /**
+     * Applies a filter to the joined tuple, with the semantics of {@link BiConstraintStream#filter(BiPredicate)}.
```

Comment:
Similar to the javadoc of that filter() method, this javadoc should mention the lack of indexing.

Reply:
In fact, the reason why I reference that other Javadoc is to avoid having to say this everywhere. That Javadoc says it, and this Javadoc says "same semantics".